File: src/courses/data/catalog/02-dsa/dsa-core/implementations/c/graphs.c
/*
 * C Implementation: Graphs (BFS & DFS Traversals)
 * Invariant: Time O(V + E), Space O(V)
 */
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int* data;
    int size;
    int capacity;
} graphs_t;

// TODO: Complete the allocation and operations
void init_graphs(graphs_t* item, int capacity) {
    item->data = (int*)malloc(capacity * sizeof(int));
    item->size = 0;
    item->capacity = capacity;
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/graphs.java
/**
 * Java Implementation: Graphs (BFS & DFS Traversals)
 * Invariant: Time O(V + E), Space O(V)
 */
package seed.dsa.core;

import java.util.*;

public class GraphsBFSDFSTraversals<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
